File: client/src/types/post-category.ts
export interface PostCategoryResponse {
    id: string
    name: string
    slug: string
    description?: string
    count: number
    icon?: string
    color?: string
}

export interface PostCategoryCreate {
    name: string
    slug: string
    description?: string
    icon?: string
    color?: string
}

export interface PostCategoryUpdate {
    name?: string
    slug?: string
    description?: string
    icon?: string
    color?: string
}

File: server/app/models/post_category.py
import uuid
from datetime import datetime, timezone

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base

def _uuid():
    return str(uuid.uuid4())


class PostCategory(Base):
    __tablename__ = "post_categories"

    id: Mapped[str] = mapped_column(String, primary_key=True, default=_uuid)
    name: Mapped[str] = mapped_column(String, nullable=False)
    slug: Mapped[str] = mapped_column(String, nullable=False, unique=True)
    description: Mapped[str] = mapped_column(String, nullable=True)
    icon: Mapped[str | None] = mapped_column(String, nullable=True)
    color: Mapped[str | None] = mapped_column(String, nullable=True)

    posts: Mapped[list["Post"]] = relationship(
        back_populates="category",
        cascade="all, delete-orphan",
        lazy="selectin"
    )
